src/engine/guidance/post_processing.cpp

```diff
@@ -528,12 +528,22 @@ bool isStaggeredIntersection(const RouteStep &previous, const RouteStep &current
     // Base decision on distance since the zig-zag is a visual clue.
     const constexpr auto MAX_STAGGERED_DISTANCE = 3; // debatable, but keep short to be on safe side
 
-    using namespace util::guidance;
+    const auto angle = [](const RouteStep &step) {
+        const auto &intersection = step.intersections.front();
+        const auto entry_bearing = intersection.bearings[intersection.in];
+        const auto exit_bearing = intersection.bearings[intersection.out];
+        return turn_angle(entry_bearing, exit_bearing);
+    };
+
+    // Instead of using turn modifiers (e.g. as in isRightTurn) we want to be more strict here.
+    // We do not want to trigger e.g. on sharp uturn'ish turns or going straight "turns".
+    // Therefore we use the turn angle to derive 90 degree'ish right / left turns.
+    // This more closely resembles what we understand as Staggered Intersection.
+    const auto is_right = [](const double angle) { return angle > 45 && angle < 135; };
+    const auto is_left = [](const double angle) { return angle > 225 && angle < 315; };
 
-    const auto left_right = isLeftTurn(previous.maneuver.instruction) && //
-                            isRightTurn(current.maneuver.instruction);
-    const auto right_left = isRightTurn(previous.maneuver.instruction) && //
-                            isLeftTurn(current.maneuver.instruction);
+    const auto left_right = is_left(angle(previous)) && is_right(angle(current));
+    const auto right_left = is_right(angle(previous)) && is_left(angle(current));
```

Comment:
We still should do an analysis of the non-obvious cases. If you, as was the case prior to this PR, allow 180 and 0 degree within the turns, strange situations might get detected.
What I am concerned about are cases like these:
![sharp-staggered](https://cloud.githubusercontent.com/assets/12932279/18264539/18c55136-7411-11e6-9b3c-3fff25f32c2b.png)

If at some point we should increase the threshold (3 meters might be too small), we could end up combining very non-obvious turns.

I can't see from the above analysis if such cases would be a valid choice to classify them as staggered. We should at least see if these cases can exist, where they are and if they don't even exist make the angle more restricted. I know I said in chat `like 45 to 135 degreeish`, but I feel that we should test these bounds.